raiden-ts/src/transport/epics/init.ts

```diff
@@ -95,23 +97,21 @@ function joinGlobalRooms(config: RaidenConfig, matrix: MatrixClient): Observable
  * @param roomIds - The ids of the rooms to filter out during sync.
  * @returns Observable of the {@link Filter} that was created.
  */
-function createFilter(matrix: MatrixClient, roomIds: string[]): Observable<Filter> {
-  return defer(() => {
-    const roomFilter = {
-      not_rooms: roomIds,
-      ephemeral: {
-        not_types: ['m.receipt', 'm.typing'],
-      },
-      timeline: {
-        limit: 0,
-        not_senders: [matrix.getUserId()!],
-      },
-    };
-    const filterDefinition = {
-      room: roomFilter,
-    };
-    return matrix.createFilter(filterDefinition);
-  });
+async function createFilter(matrix: MatrixClient, roomIds: string[]) {
```

Comment:
Are return types of async functions not annotated?

Reply:
We don't enforce annotating return types yet (we should at some point though through lint rules). Right now, we let the implicit return type on simple/obvious enough functions, or functions where extracting some internal type to put in annotation would be too complex.

Reply:
Done it as part of PR review commit. Thanks